### PR TITLE
Plugins: Improve levitate / breaking changes report in grafana/grafana

### PR DESCRIPTION
**Pull request by oshirohugo**
**What is this feature?**

This PR is adding more details to the report generated by the levitate pipeline:
- numbers of plugins affected by the breaking changes (if any)
- links to a panel where they can see each plugin affected, how many imports they are doing and the files where the imports are done (see the example of the new report in the item `Special notes for your reviewer`

**Why do we need this feature?**

This feature is necessary to give developers a better way to analyse and react to possible breaking changes.

**Who is this feature for?**

Grafana developers

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-community-team/issues/137

**Special notes for your reviewer:**

In this feature we choose to go with links instead of showing the info directly to avoid showing info of private plugins.

This is an example of the new report:
##################
<h4>Changes</h4><b>useTheme2</b><br>
<sub>/tmp/@grafana/ui@10.1.0/package/dist/index.d.ts</sub><br>
<pre lang="diff">
Parameter type changed:
-ErrorInfo
+interface ErrorInfo {
+    componentStack: string;
+}

</pre><br>
<b>GrafanaTheme2.spacing</b><br>
<sub>/tmp/@grafana/data@10.1.0/package/dist/index.d.ts</sub><br>
<pre lang="diff">
-T = unknown
+T = any

</pre><br>
<h3>Number of affected plugins: 22</h3><p>To check the plugin affected by each import, click on the links below.</p><h4>Changes</h4><a href="https://ops.grafana-ops.net/d/dmb2o0xnz/imported-property-details?orgId=1&var-propertyName=useTheme2&var-packageName=@grafana/ui">@grafana/ui/useTheme2</a><br>
<a href="https://ops.grafana-ops.net/d/dmb2o0xnz/imported-property-details?orgId=1&var-propertyName=GrafanaTheme2.spacing&var-packageName=@grafana/data">@grafana/data/GrafanaTheme2.spacing</a>

##################

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: scripts/levitate-show-affected-plugins.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        markdown += `<h4>Removals</h4>`;
+        markdown += createLinks(removals, affectingProperties).join('<br>\n');
+      }
+
+      if (changes.length > 0) {
+        markdown += `<h4>Changes</h4>`;
+        markdown += createLinks(changes, affectingProperties).join('<br>\n');
+      }
+    } else {
+      markdown += '<h4>No affected plugins found</h4>';
```
Comment (academo) [inline scripts/levitate-show-affected-plugins.js:142]: What if we don't print this part and instead just output a blank report?

My concern is that this line can give a false feeling that nothing is broken and you can go ahead and merge but in truth our database of dependency usage is not 100% reliable so we can't reliable say "nothing is affected"

Comment (oshirohugo) [reply scripts/levitate-show-affected-plugins.js:142]: Thanks for your review. It makes sense. I'll change that :D

**Review by academo — APPROVED** [review]
Excellent work on this @oshirohugo !

File: .github/workflows/detect-breaking-changes-levitate.yml [hunk 2]
```diff
@@ -134,6 +137,29 @@ jobs:
...
+          service_account: ${{ secrets.LEVITATE_SA }}
+
+      - name: 'Set up Cloud SDK'
+        uses: 'google-github-actions/setup-gcloud@v2'
+        with:
+          version: '>= 363.0.0'
+          project_id: 'grafanalabs-global'
+          install_components: 'bq'
+
+        # This is step is needed to generate a detailed levitate report
```
Comment (leventebalogh) [inline .github/workflows/detect-breaking-changes-levitate.yml:153]:
```suggestion
        # This step is needed to generate a detailed levitate report
```

File: scripts/levitate-show-affected-plugins.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    const stdout = execSync(cmd, { encoding: 'utf-8' });
+
+    const rows = stdout.trim().split('\n');
+
+    if (rows.length > 3) {
+      const pluginsColumnIndex = 3;
+      const affectedPlugins = getColumn(rows, pluginsColumnIndex);
+
+      markdown += `<h3>Number of affected plugins: ${affectedPlugins.size}</h3>`;
+      markdown += "<p>To check the plugin affected by each import, click on the links below.</p>";
```
Comment (leventebalogh) [inline scripts/levitate-show-affected-plugins.js:127]:
```suggestion
      markdown += "<p>To check the plugins affected by each import, click on the links below.</p>";
```

**Review by leventebalogh — APPROVED** [review]
Nice 🚀

**Pull request merged**